src/main/java/jenkins/plugins/office365connector/HttpWorker.java

```diff
@@ -116,6 +115,9 @@ private HttpClient getHttpClient() {
                 }
             }
         }
+        client.getParams().setConnectionManagerTimeout(timeout);
```

Comment:
what is the motivation for this change?

Reply:
so that method "getHttpClient()" would hide all httpclient creation logic - and the addition of next 2 timeouts wouldn't clutter the "run" method